tools/tests/test_osqueryd.py

```diff
@@ -190,6 +190,35 @@ def test_9_hostid_instance(self):
         self.assertTrue(daemon.isAlive())
         daemon.kill()
 
+    @test_base.flaky
+    def test_10_daemon_with_watchdog(self):
+        # Test that killed workers are respawned by the watcher
+        if os.environ.get('SANITIZE') is not None:
+            return
+        daemon = self._run_daemon({
+            "allow_unsafe": True,
+            "disable_watchdog": False,
+            "ephemeral": True,
+            "disable_database": True,
+            "disable_logging": True,
+        })
+        self.assertTrue(daemon.isAlive())
+
+        # Check that the daemon spawned a child process
+        children = daemon.getChildren()
+        self.assertTrue(len(children) > 0)
+
+        # Kill only the child worker
+        os.kill(children[0], signal.SIGINT)
+        self.assertTrue(daemon.isDead(children[0]))
+        self.assertTrue(daemon.isAlive())
+
+        # We give the daemon an opportunity to spawn the new child
+        time.sleep(10)
```

Comment:
Sure, I'll take another look at leveraging those but using those methods I wasn't able to get the test passing as it would just fall through and not wait long enough for the daemon to respawn the child, but perhaps I just need a higher `timeout` value.